src/lib/collab/channel.js

```diff
@@ -5,6 +5,20 @@ const minimumBackoff = 128 // 128ms
 const maximumBackoff = 1000 * 60 * 5 // Max 5 minutes
 const failuresBeforeCatchup = 4
 
+const resolvablePromise = () => {
+  let resolveFunction
+  const promise = new Promise(resolve => {
+    resolveFunction = resolve
+  })
+  promise.resolved = false
+  promise.resolve = function() {
+    resolveFunction()
+    promise.resolved = true
+    return promise
+  }
+  return promise
+}
```

Comment:
This should be extracted somewhere else with a js doc to explain the goal of this function

Reply:
We already have a cancelablePromise, we should add the resolvable there https://github.com/cozy/cozy-client/blob/1431dcbbe30af8b1b1eb731c065aee146d26caa4/packages/cozy-client/src/utils.js#L9

Reply:
What is the difference with Promise.resolve ?

Reply:
You can resolve later. 
It was just a local tool to avoid using two properties, one for the promise and one to externalize the resolution function.

Reply:
https://github.com/cozy/cozy-client/pull/610

But I am still unsure about the benefit of having it extracted